src/preprocess/query_generator.py

```diff
@@ -167,6 +170,160 @@ def generate_anonymize_user_queries(new_admin,old_admin, exclude_users, f, prepr
+    DELETE FROM  interpretation_comments where interpretationid in (select interpretationid from interpretation where organisationunitid in (select * from orgUnitsToDelete));
+    DELETE FROM  intepretation_likedby where interpretationid in (select interpretationid from interpretation where organisationunitid in (select * from orgUnitsToDelete));
+    DELETE FROM  interpretationuseraccesses where interpretationid in (select interpretationid from interpretation where organisationunitid in (select * from orgUnitsToDelete));
+    DELETE FROM  interpretation where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM eventvisualization_organisationunits WHERE organisationunitid      IN (SELECT * FROM orgUnitsToDelete);
+    DELETE FROM  lockexception where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  mapview_organisationunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  organisationunitattributevalues where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  program_organisationunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM programmessage_emailaddresses    WHERE programmessageemailaddressid     IN (SELECT * FROM rm_programmessage);
+    DELETE FROM programmessage_deliverychannels  WHERE programmessagedeliverychannelsid IN (SELECT * FROM rm_programmessage);
+    DELETE FROM programmessage                   WHERE id                               IN (SELECT * FROM rm_programmessage);
+    DELETE FROM  reporttable_organisationunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  userdatavieworgunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  usermembership where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  userteisearchorgunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  validationresult where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM completedatasetregistration where sourceid in (select * from orgUnitsToDelete);
+    DELETE FROM configuration WHERE selfregistrationorgunit in (select * from orgUnitsToDelete);
+    DELETE FROM minmaxdataelement WHERE sourceid in (select * from orgUnitsToDelete);
+    DELETE FROM visualization_organisationunits WHERE organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM organisationunit WHERE organisationunitid in (select * from orgUnitsToDelete);
+    DROP MATERIALIZED VIEW if exists orgUnitsToDelete  CASCADE;
+    
+    """)
+
+
+def generate_delete_org_unit_level_rules(level, f):
+    write(f,"""
+    --remove organisationUnits -- org unit
+    DROP MATERIALIZED VIEW if exists orgUnitsToDelete CASCADE;
+    """)
+    write(f,
+        " CASCADE;"
```

Comment:
From the coment above:
Right, but I change the function name by:
create_org_units_to_remove_views_and_indexes

Due i think we should identify that query come from the org_units_remove actions (we have a lot of actions in that file and only create_views_and_indixes is more hard to know what views and indixex are creating)

- About delete_org_unit_data_in_view() and delete_org_units_in_view() 

The data functions removes data and interpretations and its dependencies. 
Could be usefull have that info, i come back to read this code when someone needs now what tables are related with this action for other issues.

I will refactorize in single query as you suggest but adding a comment to identify the blocks, do you like it?


- About order:

I mantained the actual order only adding the news queries due was heavy tested removing orgunit using preprod witll aht the data, and maybe dependency issues could happear if you test in other scenary with less data.

For example maybe you test it but in a docker with 0 lockexceptions, and maybe i move this query in the past to avoid dependencies issue.

But i could follow your order, and if we find a dependencie error we could move it again. 
What do you think?

- (delete -> DELETE, etc.)
fixed

- Initial spaces:

Removed

--About f.write()

We are using: 

def write(f, text):
    print(text)
    f.write(text)
    
 
do you mean create a big string concatenating every call across all the calls and at the end decide what do with that string?
I think this is a big change to talk in next meeting and maybe do in as another issue (that format is present in all the code not only in the orgunits remover, is not related with this task).

we will refactor all the code to do it because its ussed in so many kind of situations, iterating many kind of rules, anonimization, removing orgunits... For me has sense to avoid pass the f param but maybe this change will make more harder to debug in the future (because you are looking in the output what is written line by line instead of manage a inmense string).

Reply:
> From the coment above: Right, but I change the function name by: create_org_units_to_remove_views_and_indexes
> 
> Due i think we should identify that query come from the org_units_remove actions (we have a lot of actions in that file and only create_views_and_indixes is more hard to know what views and indixex are creating)

Ok!


>     * About delete_org_unit_data_in_view() and delete_org_units_in_view()
> 
> 
> The data functions removes data and interpretations and its dependencies. Could be usefull have that info, i come back to read this code when someone needs now what tables are related with this action for other issues.
> 
> I will refactorize in single query as you suggest but adding a comment to identify the blocks, do you like it?

Yes, the distinction is hard for me to see, but of course, it's good if you have it in blocks that are meaningful!


>     * About order:
> 
> 
> I mantained the actual order only adding the news queries due was heavy tested removing orgunit using preprod witll aht the data, and maybe dependency issues could happear if you test in other scenary with less data.
> 
> For example maybe you test it but in a docker with 0 lockexceptions, and maybe i move this query in the past to avoid dependencies issue.
> 
> But i could follow your order, and if we find a dependencie error we could move it again. What do you think?

Both are fine with me. I asked to understand it better. If you feel more confident with your order, that's okay too (and now I understand why).


>     * (delete -> DELETE, etc.)
>       fixed
> 
>     * Initial spaces:
> 
> 
> Removed

Thanks!


> --About f.write()
> 
> We are using:
> 
> def write(f, text): print(text) f.write(text)
> 
> do you mean create a big string concatenating every call across all the calls and at the end decide what do with that string? I think this is a big change to talk in next meeting and maybe do in as another issue (that format is present in all the code not only in the orgunits remover, is not related with this task).
> 
> we will refactor all the code to do it because its ussed in so many kind of situations, iterating many kind of rules, anonimization, removing orgunits... For me has sense to avoid pass the f param but maybe this change will make more harder to debug in the future (because you are looking in the output what is written line by line instead of manage a inmense string).

Don't worry, it was a general comment and I do not expect us to make that change now. It was just a general idea for future reference (in case it helps, which I think it would).

So please disregard my comment if you wish.

---

The only "change" that I would find useful is, if you want me to try it too, just write down some instructions about how to test the changes. (Could I launch it with `./dhis2_clone.py config.json` for a certain example config.json`... ?)